packages/shared_preferences/shared_preferences_tool/pubspec.yaml

```diff
@@ -0,0 +1,25 @@
+name: shared_preferences_tool
+description: "DevTools extension for package:shared_preferences. Manage SharedPreferences efficiently. Edit, search, and view keys."
+publish_to: 'none'
+
+version: 1.0.0+1
+
+environment:
+  sdk: '>=3.4.0-282.1.beta <4.0.0'
```

Comment:
This should use a non-beta version now that 3.4 is in stable.

@kenzieschmoll What are the implications of shipping a package that requires 3.4 as a tool extension; do we need to make shared_preferences require the same version? (And in general, is there a requirement that the versions stay consistent between these two packages? If so we need some kind of solution for that.)

Reply:
I was using the constraints from [devtools_extensions](https://github.com/flutter/devtools/blob/master/packages/devtools_extensions/pubspec.yaml#L8). But I'll try using flutter 3.22 to see if it compiles.

Reply:
Those constraints predate 3.4.0 being released to `stable`.

I'm not sure what you mean by "see if it compiles"; if it worked on `stable` with a constraint older than `stable`, then it will also work on `stable` with a constraint equal to `stable`.

Reply:
The Dart & Flutter version used to build the DevTools extension's flutter web app (`shared_preferences_tool`) are independent of the Dart & Flutter versions required for the parent package (`shared_preferences`). These do not need to be kept in sync because the parent package `shared_preferences` will never need to version solve with `shared_preferences_tool`. When a user depends on `shared_preferences` from their app and connects to DevTools, we will load the raw web app assets from `extension/devtools/build` in an iFrame, so the Flutter version these web assets were built with shouldn't matter.

It is worth noting however, that the DevTools extension should be backwards compatible when _connecting_ to a Flutter app that is on an older version of Flutter, but this should be taken care of by the `devtools_extensions` and `devtools_app_shared` packages unless the DevTools extension is using `package:vm_service` directly in a non-backwards compatible way.